File: src/java/magma/api/text/Strings.java
package magma.api.text;

import magma.Actual;

public final class Strings {
    @Actual
    public static int length(String stripped) {
        return stripped.length();
    }

    @Actual
    public static String sliceBetween(String input, int startInclusive, int endExclusive) {
        return input.substring(startInclusive, endExclusive);
    }

    @Actual
    public static String sliceFrom(String input, int startInclusive) {
        return input.substring(startInclusive);
    }

    @Actual
    public static boolean isEmpty(String cache) {
        return cache.isEmpty();
    }

    @Actual
    public static boolean equalsTo(String left, String right) {
        return left.equals(right);
    }

    @Actual
    public static String strip(String input) {
        return input.strip();
    }

    @Actual
    public static boolean isBlank(String value) {
        return value.isBlank();
    }
}

File: src/ts/magma/api/text/Strings.ts
/*[
	Actual: magma.annotate, 
	Namespace: magma.annotate, 
	Collector: magma.api.collect, 
	EmptyHead: magma.api.collect, 
	Head: magma.api.collect, 
	List: magma.api.collect, 
	ListCollector: magma.api.collect, 
	Lists: magma.api.collect, 
	HeadedQuery: magma.api.collect.query, 
	Query: magma.api.collect.query, 
	RangeHead: magma.api.collect, 
	SingleHead: magma.api.collect, 
	Console: magma.api, 
	IOError: magma.api.io, 
	Path: magma.api.io, 
	None: magma.api.option, 
	Option: magma.api.option, 
	Some: magma.api.option, 
	Result: magma.api.result, 
	Characters: magma.api.text
]*/
import { Actual } from "../../../magma/annotate/Actual";
import { Namespace } from "../../../magma/annotate/Namespace";
export interface StringsInstance {
	length(stripped: string): number;

	sliceBetween(input: string, startInclusive: number, endExclusive: number): string;

	sliceFrom(input: string, startInclusive: number): string;

	isEmpty(cache: string): boolean;

	equalsTo(left: string, right: string): boolean;

	strip(input: string): string;

	isBlank(value: string): boolean;

}
export declare const Strings: StringsInstance;
